### ISSUE TITLE
Anoying warning on optimizations

### ISSUE THREAD

**Issue by rhoml** [elastic/curator#523]
With curator 3.4.0 I get warnings when using timestring
Elsticsearch 1.7.x
````
You are using --timestring without --older-than or --newer-than.
````

The command that I am executing is
```
curator optimize --delay 120 --max_num_segments 1 indices --prefix 'logstash-' --timestring '%Y.%W' --older-than 0 --time-unit weeks
````

**Comment by untergeek** [comment]
Ah, so I should check that `--older-than` is not empty, rather than boolean, which catches the zero as a false positive in this case.

I'll see what I can do.

**Comment by untergeek** [comment]
In the meantime, you could add `--newer-than 1040`, which would effectively add a null filter that includes all weekly indices newer than 20 years in the future.  That will remove the warning. :smile:

**Comment by rhoml** [comment]
Yes and no. 

What I want to accomplish here is to optimize all indexes older than 1 day but actually `older-than 1` means 2 days older which is not the desired behaviour I think.

**Comment by untergeek** [comment]
Well, you can't do days with your time unit as weeks. 

The functionality has two things to keep in mind. 

1. It counts days from 00:00 UTC to 00:00 UTC the next day
2. Weeks are counted slightly more awkwardly because of differences in calculation. That 2 week thing is the result.

**Comment by rhoml** [comment]
sorry I confused I meant +1 week so if my weekly indices is named logstash-2015.40 and logstash-2015.41 I would like to be able to optimize the first one.

**Comment by untergeek** [comment]
@rhoml Yes, I apologize for the awkwardness, but this is because the way calculations are done, you can't accurately calculate 1 week old until the current week has become 1 _full_ week, which results in `--older-than 1` not acting until the week is basically 2 weeks old.  This is why `--older-than 0` was added, so this oddity can be worked around. 

The `--newer-than 1040` workaround will still quiet the warning, while it should yield the desired result.

**Comment by rhoml** [comment]
I am afraid it doesn't respect the newer-than 1040 =/ it keeps optimizing all of my indexes.

**Comment by untergeek** [comment]
In order to accurately reproduce this, can you send me the list of index names you were trying to work on?  I can attempt to create a test to replicate this behavior.

**Comment by rhoml** [comment]
in our case is logstash_environment_service-%Y.%W for example

**Comment by untergeek** [comment]
I guess I can try to generate some of those.  How many do you have?

**Comment by untergeek** [comment]
This should be addressed by #595 and #596 with Curator 4.0 imminent.

If you feel this is in error, please feel free to reopen this ticket or open another.
